pkg/placement/what-resolver_test.go

```diff
@@ -126,36 +122,39 @@ func TestWhatResolver(t *testing.T) {
-	setFakeClusterAPIResources(fakeKubeClusterClientset.Fake, wds1N.Path(), []*metav1.APIResourceList{
-		{GroupVersion: k8sCoreGroupVersion.String(),
-			APIResources: []metav1.APIResource{nsResource, cmResource},
-		}})
-	fcd := FakeClusterDisco{fakeKubeClusterClientset.Discovery(), fakeKubeClusterClientset}
+	// fakeKubeClusterClientset := fakekube.NewSimpleClientset(ns1, cm1)
+	// k8sCoreGroupVersion := metav1.GroupVersion{Version: "v1"}
+	// usualVerbs := []string{"get", "list", "watch"}
+	// nsResource := metav1.APIResource{Name: "namespaces", SingularName: "namespace", Namespaced: false, Version: "v1", Kind: "Namespace", Verbs: usualVerbs}
+	// cmResource := metav1.APIResource{Name: "configmaps", SingularName: "configmap", Namespaced: true, Version: "v1", Kind: "ConfigMap", Verbs: usualVerbs}
+	// setFakeClusterAPIResources(fakeKubeClusterClientset.Fake, wds1N.Path(), []*metav1.APIResourceList{
+	// 	{GroupVersion: k8sCoreGroupVersion.String(),
+	// 		APIResources: []metav1.APIResource{nsResource, cmResource},
+	// 	}})
+	//fcd := FakeClusterDisco{fakeKubeClusterClientset.Discovery(), fakeKubeClusterClientset}
 	scheme := machruntime.NewScheme()
 	orDie(apiextensionsv1.AddToScheme(scheme))
 	orDie(kcpapisv1alpha1.AddToScheme(scheme))
 	orDie(k8scorev1.AddToScheme(scheme))
-	fakeKCPClient := fakeclusterkcp.NewSimpleClientset()
-	bindingFactory := bindingfactory.NewSharedInformerFactory(fakeKCPClient, 0)
-	bindingClusterPreInformer := bindingFactory.Apis().V1alpha1().APIBindings()
+	//fakeKCPClient := fakeclusterkcp.NewSimpleClientset()
+	//bindingFactory := bindingfactory.NewSharedInformerFactory(fakeKCPClient, 0)
+	//bindingClusterPreInformer := bindingFactory.Apis().V1alpha1().APIBindings()
 	fakeDynamicClusterClientset := fakeclusterdynamic.NewSimpleDynamicClient(scheme, ns1, cm1, ns2, cm2, cm3)
 	dynamicClusterInformerFactory := clusterdynamicinformer.NewDynamicSharedInformerFactory(fakeDynamicClusterClientset, 0)
 	// crdClusterPreInformer := dynamicClusterInformerFactory.ForResource(apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions"))
 	// bindingClusterPreInformer := dynamicClusterInformerFactory.ForResource(kcpapisv1alpha1.SchemeGroupVersion.WithResource("apibindings"))
 
-	fakeApiExtClusterClientset := fakeapiext.NewSimpleClientset()
-	apiExtClusterFactory := apiextinfact.NewSharedInformerFactory(fakeApiExtClusterClientset, 0)
-	crdClusterPreInformer := apiExtClusterFactory.Apiextensions().V1().CustomResourceDefinitions()
+	//fakeApiExtClusterClientset := fakeapiext.NewSimpleClientset()
+	//apiExtClusterFactory := apiextinfact.NewSharedInformerFactory(fakeApiExtClusterClientset, 0)
+	//crdClusterPreInformer := apiExtClusterFactory.Apiextensions().V1().CustomResourceDefinitions()
 
 	fakeUpKubeClient := fakeupkube.NewSimpleClientset()
 	kbSpaceRelation := kbuser.NewKubeBindSpaceRelation(ctx, fakeUpKubeClient)
+	spaceProviderNs := "spaceprovider-fake"
+	// TODO fake
```

Comment:
There is no framework or exemplar for fake `*rest.Config`. If #1271 were merged and used here, then fake clientsets could be used instead.